hudi-spark-datasource/hudi-spark3.0.x/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/Spark30ParquetReader.scala

```diff
@@ -174,7 +190,7 @@ class Spark30ParquetReader(enableVectorizedReader: Boolean,
       reader.initialize(split, hadoopAttemptContext)
 
       val fullSchema = requiredSchema.toAttributes ++ partitionSchema.toAttributes
-      val unsafeProjection = GenerateUnsafeProjection.generate(fullSchema, fullSchema)
+      val unsafeProjection = schemaEvolutionUtils.generateUnsafeProjection(fullSchema, timeZoneId)
```

Comment:
The legacy parquet file format implementation checks `shouldAppendPartitionValues` for the logic below.  Is that no longer needed?
<img width="898" alt="Screenshot 2024-04-12 at 18 02 23" src="https://github.com/apache/hudi/assets/2497195/611e89eb-0f7d-4f4b-876b-6f2583996332">